infrastructure/template.yaml

```diff
@@ -517,6 +517,260 @@ Resources:
+    Metadata:
+      BuildMethod: esbuild
+      BuildProperties:
+        Minify: true
+        Target: "es2020"
+        Sourcemap: true
+        EntryPoints:
+          - delete-user-account-passthrough.ts
+
+  UserAccountDeletionTopic:
+    Type: AWS::SNS::Topic
+    Properties:
+      DisplayName: UserAccountDeletionTopic
+      KmsMasterKeyId: !Ref SnsKmsKey
+      Subscription:
+        - Endpoint:
+            Fn::GetAtt:
+              - "UserServicesToDeleteQueue"
+              - "Arn"
+          Protocol: "sqs"
+      TopicName: UserAccountDeletion
+
+  UserServicesToDeleteQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      KmsMasterKeyId: !GetAtt QueueKmsKey.Arn
+    DeletionPolicy: Delete
+    UpdateReplacePolicy: Delete
+
+  DeleteUserServicesFunction:
+    DependsOn:
+      - DeleteUserServicesFunctionLogGroup
+    Type: AWS::Serverless::Function
+    Properties:
+      Architectures: ["x86_64"]
+      CodeUri: ../lambda/delete-user-services/
+      Events:
+        UserServicesToDeleteQueue:
+          Type: SQS
+          Properties:
+            Queue: !GetAtt UserServicesToDeleteQueue.Arn
+      DeadLetterQueue:
+        Type: SQS
+        TargetArn: !GetAtt DeleteUserServicesDeadLetterQueue.Arn
+      Handler: write-user-services.handler
+      KmsKeyArn: !GetAtt LambdaKMSKey.Arn
+      PackageType: Zip
+      MemorySize: 128
+      Runtime: nodejs16.x
+      Role: !GetAtt DeleteUserServicesRole.Arn
+      Timeout: 5
+      Environment:
+        Variables:
+          TABLE_NAME: !Ref UserServicesStoreTableName
+          DLQ_URL: !Ref DeleteUserServicesDeadLetterQueue
+    Metadata:
+      BuildMethod: esbuild
+      BuildProperties:
+        Minify: true
+        Target: "es2020"
+        Sourcemap: true
+        EntryPoints:
+          - write-user-services.ts
+
+  UserDeleteAccountPassthroughRole:
+    Type: AWS::IAM::Role
+    Properties:
+      PermissionsBoundary: !If
+        - UsePermissionsBoundary
+        - !Ref PermissionsBoundary
+        - !Ref AWS::NoValue
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+        - !Ref UserDeleteAccountPassthroughPolicy
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action:
+              - "sts:AssumeRole"
+
+  UserDeleteAccountPassthroughPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Action:
+              - sqs:ReceiveMessage
+              - sqs:DeleteMessage
+              - sqs:GetQueueAttributes
+            Resource: !GetAtt UserDeleteAccountQueue.Arn
+          - Effect: Allow
+            Action:
+              - sns:Publish
+            Resource: !Ref UserAccountDeletionTopic
```

Comment:
This is odd but couldn't get it to like a `!GetAtt UserAccountDelegationTopic.Arn`
Not really sure why.

Reply:
Alex solved that one, the docs show that yeah doesn't support ARN.
So this is just another strange cloudformation inconsistency:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sns-topic.html#aws-resource-sns-topic-return-values-fn--getatt